src/runway/sources/git.py

```diff
@@ -0,0 +1,114 @@
+"""'Git' type Path Source."""
```

Comment:
Can we add documentation here to be pulled in by `automodule`? It can then be added to `docs/source/runway_config.rst` under the `Module.Path` header unless you can think of a better place to put it.

It should describe the different components of the syntax for defining a git source and show examples.